### PR TITLE
Patch CAS Active scope respecting active_homeless_status_override setting

### PR DESCRIPTION
**Pull request by gigxz**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Follow-up to https://github.com/greenriver/hmis-warehouse/pull/4263 to include the same logic in the cas_active scope.

## Type of change
- [x] Bug fix

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: spec/models/grda_warehouse/cas_active_spec.rb [hunk 1]
```diff
@@ -69,12 +73,15 @@
     it 'finds one client who is active for CAS' do
       travel_to Time.local(2016, 2, 15) do
         expect(GrdaWarehouse::Hud::Client.destination.map(&:active_in_cas?).count(true)).to eq(1)
+        # Fails, cas_active scope does not respect ineligible_uses_extrapolated_days setting
+        # expect(GrdaWarehouse::Hud::Client.destination.cas_active.count).to eq(1)
```
Comment (gigxz) [inline spec/models/grda_warehouse/cas_active_spec.rb]: Not sure if this is a bug or on purpose

Comment (eanders) [reply spec/models/grda_warehouse/cas_active_spec.rb]: I have a fix for this, but for some reason it's causing and issue with a different test and the transactions.  So forthcoming....

**Review by eanders — APPROVED** [review]
Pending passing tests.

**Comment by gigxz** [comment]
Thanks @eanders

**Pull request merged**